Add use of names and remove initial participation limit

josephmckinsey opened the pull request:
The previous version required the user to preselect the number of
participants. This became annoying when someone new enters the room.
Now you specify the stopping point for the algorithm at each stage.

 * Query at beginning of use for names. Optional use.
 * Keep the structure of the program as close as possible to the original, so
   the program uses a few if statements with useNames as a conditional.
 * Use a dictionary to define a number to a name.
    * This also applies to using numbers, for less bias, since the number of
      people in the index increased.
 * Create better checking for y/n along with default blank option capitalized
 * Lists winners at end of game.

josephmckinsey committed:
Add use of names and remove initial participation limit

The previous version required the user to preselect the number of
participants. This became annoying when someone new enters the room.
Now you specify the stopping point for the algoirthm at each stage.

 * Query at beginning of use for names. Optional use.
 * Keep the structure of the program as close as possible to the original, so
   the program uses a few if statements with useNames as an operation.
 * Use a dictionary to define a number to a name.
    * This also applies to using numbers, for less bias, since the number of
      people in the index increased.
 * Create better checking for y/n along with default blank option capitalized
 * Lists winners at end of game.

jackrosenthal commented:
Thanks for the PR! Can you clean up the code a little? Couple of things:

- Make the code style match PEP-8 if you can
- Don't use variable names like `temp`
- There's lots of yes/no prompts, can you simplify that to either a function or somehow otherwise remove the repeated code?

josephmckinsey committed:
Clean code and sane structuring

Now uses code commenting, doc strings, and functions. The previous
version used some while loops and unimportant temporary variables.

    * Add function to determine whether to stop.
    * Add function with y/n input.
    * Change while loops use condition instead of breakpoint.
    * Add doc strings to helper functions.
    * Clean code for printing winners.
    * Use more meaningful counter variables instead of e and s.
    * Add commenting.
    * Use __name__ == "__main__" and a main function.

josephmckinsey commented:
OK. How is it now?
